Assignment2/Q2.java

```diff
@@ -0,0 +1,180 @@
+class Q2
+{
+	public static void main(String[] args)
+	{
+		BinaryTree tree = new BinaryTree(16);
+		tree.add(9);
+		tree.add(18);
+		tree.add(19);
+		tree.add(3);
+		tree.add(14);
+		tree.add(1);
+		tree.add(5);
+
+		Node x = lowest(tree.root, new Node(19), new Node(4));
+		if(x != null)
+		{
+			System.out.println("COMMOM: " + x.getValue());
+		}
+		
+	}
+
+	//checks if both nodes are actually present in the tree 
+	public static Node lowest(Node root, Node p, Node q)
+	{
+		if(findHelp(root, p.getValue()) != 0 && findHelp(root, q.getValue()) != 0)
+		{
+			return lowestCommonAncestor(root, p, q);
+		}
+		else
+		{
+			System.out.println("Both of the nodes do not exist");
+			return null;
+		}
+	}
+
+	//find the lowest common ancestor (recurviely) 
+	public static Node lowestCommonAncestor(Node root, Node p, Node q) 
+	{
+		if(root==null)
+		{
+        	return null;
+		}
+ 
+    	if(root.getValue()==p.getValue() || root.getValue()==q.getValue())
+    	{
+        	return root;
+        }
+ 
+    	Node l = lowestCommonAncestor(root.left, p, q);
+    	Node r = lowestCommonAncestor(root.right, p, q);
+ 
+    	if(l != null && r != null)
+    	{
+        	return root;
+    	}
+    	else if(l == null && r == null)
+    	{ 
+        	return null;
+    	}
+    	else
+    	{
+        	return l==null?r:l;
```

Comment:
This method would not execute if 1 or both of the nodes aren't in the tree.

In the method 'lowest' I test to see if both are in the tree first - only if both are in the tree will this helper method run